File: apps/android/src/main/kotlin/space/bitos/app/ui/static/StaticPagesScreen.kt
package space.bitos.app.ui.static

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.foundation.background
import androidx.compose.runtime.getValue
import androidx.compose.runtime.setValue
import space.bitos.app.ui.theme.BitOSColors
import space.bitos.app.ui.theme.BitOSSpacing
import space.bitos.core.settings.StaticPagesContent
import space.bitos.core.settings.StaticSection

/**
 * APP-020 static pages (spec §3.20): About · Privacy · Terms — the
 * complete legacy copy from shared `StaticPagesContent`, rendered as
 * section cards (legacy Flutter static_pages parity).
 */
@Composable
fun StaticPagesScreen(
    initialPage: String = "about",
    onClose: () -> Unit,
) {
    var page by androidx.compose.runtime.saveable.rememberSaveable { androidx.compose.runtime.mutableStateOf(initialPage) }

    Column(
        Modifier
            .fillMaxSize()
            .background(BitOSColors.background)
            ,
    ) {
        // Page switcher.
        Row(
            Modifier
                .fillMaxWidth()
                .padding(horizontal = BitOSSpacing.screen, vertical = BitOSSpacing.sm),
            horizontalArrangement = Arrangement.spacedBy(BitOSSpacing.md),
        ) {
            listOf("about", "privacy", "terms").forEach { key ->
                val selected = page == key
                val label = when (key) {
                    "privacy" -> "Privacy"
                    "terms" -> "Terms"
                    else -> "About"
                }
                TextButton(onClick = { page = key }) {
                    Text(
                        label,
                        fontWeight = if (selected) FontWeight.W800 else FontWeight.W600,
                        color = if (selected) BitOSColors.primary else BitOSColors.textSecondary,
                    )
                }
            }
            Spacer(Modifier.weight(1f))
            TextButton(onClick = onClose) { Text("Close", color = BitOSColors.textSecondary) }
        }

        Column(
            Modifier
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
                .padding(horizontal = BitOSSpacing.screen)
                .padding(bottom = BitOSSpacing.xl),
            verticalArrangement = Arrangement.spacedBy(BitOSSpacing.md),
        ) {
            when (page) {
                "privacy" -> PrivacyPage()
                "terms" -> TermsPage()
                else -> AboutPage()
            }
        }
    }
}

@Composable
private fun AboutPage() {
    // Hero.
    SectionCard {
        Column(Modifier.padding(BitOSSpacing.base), horizontalAlignment = Alignment.CenterHorizontally) {
            Text(
                StaticPagesContent.ABOUT_HEADLINE,
                style = MaterialTheme.typography.headlineLarge,
                fontWeight = FontWeight.W900,
                color = BitOSColors.primary,
            )
            Text(
                StaticPagesContent.ABOUT_SUBTITLE,
                style = MaterialTheme.typography.labelMedium,
                color = BitOSColors.textTertiary,
            )
            Spacer(Modifier.height(BitOSSpacing.sm))
            Text(
                StaticPagesContent.ABOUT_HERO_BODY,
                style = MaterialTheme.typography.bodyMedium,
                color = BitOSColors.textSecondary,
            )
        }
    }
    SectionCard(StaticPagesContent.WHAT_IS_NOSTR_TITLE, StaticPagesContent.WHAT_IS_NOSTR_BODY)
    SectionCard(StaticPagesContent.OPEN_SOURCE_TITLE, StaticPagesContent.OPEN_SOURCE_BODY + "\n" + StaticPagesContent.OPEN_SOURCE_LAUNCH)
    // Features.
    Text("What makes BitOS different", style = MaterialTheme.typography.titleMedium, fontWeight = FontWeight.W800)
    StaticPagesContent.aboutFeatures.forEach { feature ->
        SectionCard(feature.title, feature.body)
    }
    // Legal link.
    Text(
        "By using BitOS you agree to our Terms and Privacy Policy.",
        style = MaterialTheme.typography.labelSmall,
        color = BitOSColors.textTertiary,
    )
}

@Composable
private fun PrivacyPage() {
    SectionCard {
        Column(Modifier.padding(BitOSSpacing.base)) {
            Text(StaticPagesContent.PRIVACY_TITLE, style = MaterialTheme.typography.headlineSmall, fontWeight = FontWeight.W800)
            Text(StaticPagesContent.PRIVACY_UPDATED, style = MaterialTheme.typography.labelSmall, color = BitOSColors.textTertiary)
            Spacer(Modifier.height(BitOSSpacing.sm))
            Text(StaticPagesContent.PRIVACY_INTRO, style = MaterialTheme.typography.bodyMedium, color = BitOSColors.textSecondary)
        }
    }
    SectionCard(StaticPagesContent.PRIVACY_SUMMARY_TITLE, StaticPagesContent.PRIVACY_SUMMARY_BODY, highlight = true)
    StaticPagesContent.privacySections.forEach { section ->
        SectionCard(section.title, section.body)
    }
}

@Composable
private fun TermsPage() {
    SectionCard {
        Column(Modifier.padding(BitOSSpacing.base)) {
            Text(StaticPagesContent.TERMS_TITLE, style = MaterialTheme.typography.headlineSmall, fontWeight = FontWeight.W800)
            Text(StaticPagesContent.TERMS_UPDATED, style = MaterialTheme.typography.labelSmall, color = BitOSColors.textTertiary)
            Spacer(Modifier.height(BitOSSpacing.sm))
            Text(StaticPagesContent.TERMS_INTRO, style = MaterialTheme.typography.bodyMedium, color = BitOSColors.textSecondary)
        }
    }
    SectionCard(StaticPagesContent.TERMS_SUMMARY_TITLE, StaticPagesContent.TERMS_SUMMARY_BODY, highlight = true)
    StaticPagesContent.termsSections.forEach { section ->
        SectionCard(section.title, section.body)
    }
}

@Composable
private fun SectionCard(title: String, body: String, highlight: Boolean = false) {
    SectionCard(highlight = highlight) {
        Column(Modifier.padding(BitOSSpacing.base)) {
            Text(title, style = MaterialTheme.typography.titleSmall, fontWeight = FontWeight.W700)
            Spacer(Modifier.height(BitOSSpacing.xs))
            Text(body, style = MaterialTheme.typography.bodyMedium, color = BitOSColors.textSecondary)
        }
    }
}

@Composable
private fun SectionCard(highlight: Boolean = false, content: @Composable () -> Unit) {
    Surface(
        shape = RoundedCornerShape(14.dp),
        color = if (highlight) BitOSColors.primaryContainer else BitOSColors.surface,
        modifier = Modifier.fillMaxWidth(),
    ) {
        content()
    }
}

File: apps/ios/BitOS/Features/Static/StaticPagesScreen.swift
import BusinessCore
import SwiftUI

/**
 * APP-020 static pages (spec §3.20): About · Privacy · Terms — the
 * complete legacy copy from shared `StaticPagesContent` through the
 * bridge, rendered as section cards (legacy Flutter static_pages parity).
 */
struct StaticPagesScreen: View {
    @State var page: String
    let onClose: () -> Void
    private let bridge = BusinessCoreBridge()

    init(initialPage: String = "about", onClose: @escaping () -> Void) {
        _page = State(initialValue: initialPage)
        self.onClose = onClose
    }

    var body: some View {
        VStack(spacing: 0) {
            HStack(spacing: BitOSTheme.Spacing.md) {
                ForEach(["about", "privacy", "terms"], id: \.self) { key in
                    Button(label(for: key)) { page = key }
                        .font(.system(size: 13, weight: page == key ? .heavy : .semibold))
                        .foregroundStyle(page == key ? BitOSTheme.accent : BitOSTheme.textSecondary)
                }
                Spacer()
                SheetCloseButton(action: onClose)
            }
            .padding(.horizontal, BitOSTheme.Spacing.screen)
            .padding(.vertical, BitOSTheme.Spacing.sm)

            ScrollView {
                VStack(spacing: BitOSTheme.Spacing.md) {
                    switch page {
                    case "privacy": privacyPage
                    case "terms": termsPage
                    default: aboutPage
                    }
                }
                .padding(.horizontal, BitOSTheme.Spacing.screen)
                .padding(.bottom, BitOSTheme.Spacing.xl)
            }
        }
        .background(BitOSTheme.background)
        .preferredColorScheme(BitOSTheme.preferredScheme)
    }

    private func label(for key: String) -> String {
        switch key {
        case "privacy": return "Privacy"
        case "terms": return "Terms"
        default: return "About"
        }
    }

    // ── About ───────────────────────────────────────────────────

    @ViewBuilder
    private var aboutPage: some View {
            HeroCard {
                VStack(spacing: BitOSTheme.Spacing.sm) {
                    Text(bridge.staticAboutHeadline())
                        .font(.system(size: 32, weight: .black))
                        .foregroundStyle(BitOSTheme.accent)
                    Text(bridge.staticAboutSubtitle())
                        .font(.system(size: 12))
                        .foregroundStyle(BitOSTheme.textTertiary)
                    Text(bridge.staticAboutHeroBody())
                        .font(.system(size: 14))
                        .foregroundStyle(BitOSTheme.textSecondary)
                        .multilineTextAlignment(.center)
                }
            }
            ForEach(Array(bridge.staticAboutFeatures().enumerated()), id: \.offset) { _, entry in
                SectionCard(title: entry["title"] as? String ?? "", body: entry["body"] as? String ?? "")
            }
    }

    // ── Privacy ─────────────────────────────────────────────────

    @ViewBuilder
    private var privacyPage: some View {
            HeaderCard(title: bridge.staticPrivacyTitle(), updated: bridge.staticPrivacyUpdated(), intro: bridge.staticPrivacyIntro())
            SectionCard(title: bridge.staticPrivacySummaryTitle(), body: bridge.staticPrivacySummaryBody(), highlight: true)
            ForEach(Array(bridge.staticPrivacySections().enumerated()), id: \.offset) { _, entry in
                SectionCard(title: entry["title"] as? String ?? "", body: entry["body"] as? String ?? "")
            }
    }

    // ── Terms ───────────────────────────────────────────────────

    @ViewBuilder
    private var termsPage: some View {
            HeaderCard(title: bridge.staticTermsTitle(), updated: bridge.staticTermsUpdated(), intro: bridge.staticTermsIntro())
            SectionCard(title: bridge.staticTermsSummaryTitle(), body: bridge.staticTermsSummaryBody(), highlight: true)
            ForEach(Array(bridge.staticTermsSections().enumerated()), id: \.offset) { _, entry in
                SectionCard(title: entry["title"] as? String ?? "", body: entry["body"] as? String ?? "")
            }
    }

    // ── Pieces ──────────────────────────────────────────────────

    private struct HeaderCard: View {
        let title: String
        let updated: String
        let intro: String

        var body: some View {
            HeroCard {
                VStack(alignment: .leading, spacing: BitOSTheme.Spacing.xs) {
                    Text(title).font(.system(size: 22, weight: .heavy))
                    Text(updated).font(.system(size: 11)).foregroundStyle(BitOSTheme.textTertiary)
                    Text(intro).font(.system(size: 13)).foregroundStyle(BitOSTheme.textSecondary)
                }
            }
        }
    }

    private struct SectionCard: View {
        var title: String? = nil
        let bodyText: String
        var highlight = false

        var body: some View {
            VStack(alignment: .leading, spacing: BitOSTheme.Spacing.xs) {
                if let title {
                    Text(title).font(.system(size: 15, weight: .bold))
                }
                Text(bodyText)
                    .font(.system(size: 13))
                    .foregroundStyle(BitOSTheme.textSecondary)
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding(BitOSTheme.Spacing.base)
            .background(
                RoundedRectangle(cornerRadius: 14, style: .continuous)
                    .fill(highlight ? BitOSTheme.accentContainer : BitOSTheme.surface)
            )
        }

        init(title: String? = nil, body text: String, highlight: Bool = false) {
            self.title = title
            self.bodyText = text
            self.highlight = highlight
        }
    }

    private struct HeroCard<Content: View>: View {
        @ViewBuilder let content: () -> Content

        var body: some View {
            content()
                .frame(maxWidth: .infinity)
                .padding(BitOSTheme.Spacing.base)
                .background(
                    RoundedRectangle(cornerRadius: 14, style: .continuous)
                        .fill(BitOSTheme.surface)
                )
        }
    }
}
